lmdk/cmake/xtensa-toolchain.cmake

```diff
@@ -0,0 +1,25 @@
+# Xtensa CMake toolchain file. Apply it using CMAKE_TOOLCHAIN_FILE variable.
+
+set(CMAKE_SYSTEM_NAME Generic)
+set(CMAKE_SYSTEM_VERSION 1)
+
+cmake_path(CONVERT $ENV{XTENSA_TOOLCHAIN_PATH} TO_CMAKE_PATH_LIST XTENSA_TOOLCHAIN_PATH)
+set(TOOLCHAIN_BASE ${XTENSA_TOOLCHAIN_PATH}/XtensaTools)
+set(CROSS_COMPILE ${TOOLCHAIN_BASE}/bin/xt-)
+
+# clang or xcc
+find_program(CMAKE_C_COMPILER NAMES ${CROSS_COMPILE}xcc NO_DEFAULT_PATH)
+# clang++ or xc++
+find_program(CMAKE_CXX_COMPILER NAMES ${CROSS_COMPILE}xc++ NO_DEFAULT_PATH)
+
+find_program(CMAKE_LD NAMES ${CROSS_COMPILE}ld NO_DEFAULT_PATH)
+find_program(CMAKE_AR NAMES ${CROSS_COMPILE}ar NO_DEFAULT_PATH)
+find_program(CMAKE_RANLIB NAMES ${CROSS_COMPILE}ranlib NO_DEFAULT_PATH)
+find_program(CMAKE_OBJCOPY NAMES ${CROSS_COMPILE}objcopy NO_DEFAULT_PATH)
+find_program(CMAKE_OBJDUMP NAMES ${CROSS_COMPILE}objdump NO_DEFAULT_PATH)
+find_program(CMAKE_NM NAMES ${CROSS_COMPILE}nm NO_DEFAULT_PATH)
+
+set(CMAKE_FIND_ROOT_PATH ${TOOLCHAIN_BASE}/xtensa-elf)
+set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
```

Comment:
What is the purpose of those?

Reply:
These are path to sysroot for cross-compilation and settings how to search for execs, headers and libraries. CMake uses these settings to alter find_library() behaviour. These here are set to search only in sysroot provided by toolchain and never search in host sysroot. For example, if find_library() is called to search for e.g. libm.a, without these settings it can try to find libm.a in your host GCC installation but we need libm.a that comes with cross-platform toolchain.